src/components/learner-credit-management/data/utils.js

```diff
@@ -337,3 +337,57 @@ export async function retrieveBudgetDetailActivityOverview({
   }
   return result;
 }
+
+/**
+ * Takes the raw selected flat rows data from the 'Assigned' datatable and returns metadata that is used for tracking
+ * bulk enrollment of reminders and bulk enrollment of cancellations.
+ * @param {Array} selectedFlatRows An array of selectedFlatRows from the activity 'Assigned' table
+ * @returns {{
+ * uniqueLearnerState: [String],
+ * totalContentQuantity: Number,
+ * assignmentConfigurationUuid: String,
+ * assignmentUuids: [String]
+ * uniqueContentKeys: [String],
+ * uniqueAssignmentState: [String],
+ * totalSelectedRows: Number,
+ * }}
+ */
+export const transformSelectedRows = (selectedFlatRows) => {
+  const assignmentUuids = selectedFlatRows.map(item => item.id);
+  const totalSelectedRows = selectedFlatRows.length;
+
+  // list of unique courses
```

Comment:
nit: if the result of this section below is to generate `uniqueLearnerState` (object), should the comment reflect that its creating a map of how many selected per course key? At first read, based on this comment, I was expecting the logic to generate a list of unique strings as opposed to an object keeping track of how many times each course key was selected.

Same feedback for the other sections (i.e., "list of unique learner states", "list of unique assignment states", etc.).